File: api/src/Commerce.Domain/Users/Address.cs
namespace Commerce.Domain.Users;

public class Address
{
    public int Id { get; set; }
    public Guid UserId { get; set; }

    public string Title { get; set; } = null!;
    public string FullName { get; set; } = null!;
    public string Phone { get; set; } = null!;
    public string City { get; set; } = null!;
    public string District { get; set; } = null!;
    public string FullAddress { get; set; } = null!;
    public bool IsDefault { get; set; }
}

File: web/src/lib/validations/address.ts
import { z } from "zod";

// .NET SaveAddressRequestValidator ile birebir (OrderValidators.cs).
export const addressFormSchema = z.object({
	title: z.string().min(1, "Başlık gerekli.").max(50),
	fullName: z.string().min(1, "Ad soyad gerekli.").max(200),
	phone: z
		.string()
		.min(1, "Telefon gerekli.")
		.max(30)
		.regex(/^[0-9\s+\-()]+$/, "Telefon numarası geçersiz."),
	city: z.string().min(1, "İl gerekli.").max(100),
	district: z.string().min(1, "İlçe gerekli.").max(100),
	fullAddress: z
		.string()
		.min(10, "Adres en az 10 karakter olmalı.")
		.max(1000),
	isDefault: z.boolean(),
});
export type AddressFormInput = z.infer<typeof addressFormSchema>;
